qiita_ware/processing_pipeline.py

```diff
@@ -22,51 +22,56 @@
 
 def _get_qiime_minimal_mapping(prep_template, out_dir):
     """Generates a minimal QIIME-compliant mapping file for split libraries
-
     The columns of the generated file are, in order: SampleID, BarcodeSequence,
     LinkerPrimerSequence, Description. All values are taken from the prep
     template except for Description, which always receive the value "Qiita MMF"
-
     Parameters
     ----------
     prep_template : PrepTemplate
         The prep template from which we need to generate the minimal mapping
     out_dir : str
         Path to the output directory
-
     Returns
     -------
     list of str
         The paths to the qiime minimal mapping files
     """
     from functools import partial
+    from collections import defaultdict
     from os.path import join
     import pandas as pd
 
-    # Get the data in a pandas DataFrame, so it is easier to manage
-    pt = prep_template.to_dataframe()
+    # The prep templates has a QIIME mapping file, get it
+    qiime_map = pd.read_csv(prep_template.qiime_map_fp, sep='\t',
+                            keep_default_na=False, na_values=['unknown'],
+                            index_col=False,
+                            converters=defaultdict(lambda: str))
+    qiime_map.set_index('#SampleID', inplace=True, drop=True)
 
-    # We now need to rename some columns to be QIIME compliant.
-    # Hopefully, this conversion won't be needed if QIIME relaxes its
-    # constraints
-    pt.rename(columns={'barcodesequence': 'BarcodeSequence',
-                       'linkerprimersequence': 'LinkerPrimerSequence'},
-              inplace=True)
-    pt['Description'] = pd.Series(['Qiita MMF'] * len(pt.index),
-                                  index=pt.index)
+    # We use our own description to avoid potential processing problems
+    qiime_map['Description'] = pd.Series(['Qiita MMF'] * len(qiime_map.index),
+                                         index=qiime_map.index)
 
     # We ensure the order of the columns as QIIME is expecting
     cols = ['BarcodeSequence', 'LinkerPrimerSequence', 'Description']
 
-    # If the study has more than 1 lane, we should generate a qiita MMF for
-    # each of the lanes. We know how to split the prep template based on
-    # the run_prefix column
-    output_fps = []
     path_builder = partial(join, out_dir)
-    for prefix, df in pt.groupby('run_prefix'):
-        df = df[cols]
-        out_fp = path_builder("%s_MMF.txt" % prefix)
-        output_fps.append(out_fp)
+    if 'run_prefix' in qiime_map:
```

Comment:
I don't recall from the stack of pull requests that have been coming in, but there are tests for both of these two branches right? i.e. cases with run_prefix and cases without it.